### PR TITLE
feat(tasks): add recurring tasks (daily/weekly/monthly) with next occurrence on completion

### PR DESCRIPTION
**Pull request by ethanzrd**
# feat(tasks): add recurring tasks (daily/weekly/monthly) with next occurrence on completion

## Summary

This PR adds support for recurring tasks to the TodoApp. Users can now select daily, weekly, or monthly recurrence options when creating or editing tasks. When a recurring task is marked complete, the system automatically creates the next occurrence with the due date shifted by the chosen interval.

**Key Changes:**
- Extended `Task` interface with optional `recurrence` field
- Added `computeNextDueDate` utility for date calculations (handles monthly edge cases)  
- Updated AddTask and EditTask UIs with recurrence selector dropdowns
- Modified task completion logic (both single and bulk) to spawn next occurrences
- Maintains backward compatibility - existing tasks without recurrence continue to work unchanged

## Review & Testing Checklist for Human

- [ ] **Test date calculation edge cases**: Try monthly recurrence with tasks due on the 31st (should handle Feb 28/29 correctly), leap years, etc.
- [ ] **Verify next occurrence creation**: Mark recurring tasks complete and confirm the new task has correct name, deadline, color, categories, etc. and `done=false`
- [ ] **Test UI integration**: Create and edit tasks with recurrence options - verify the dropdowns look consistent with existing design
- [ ] **Test bulk completion**: Select multiple recurring tasks and mark them done - should create next occurrences for all of them
- [ ] **Backward compatibility**: Confirm existing tasks (without recurrence) still work normally and aren't affected

**Recommended Test Plan:**
1. Create daily/weekly/monthly recurring tasks with and without deadlines
2. Mark them complete individually and in bulk  
3. Verify next occurrences appear with correct dates and properties
4. Edit existing tasks to add/change recurrence settings
5. Test edge cases like monthly tasks on the 31st day

---

### Diagram

```mermaid
%%{ init : { "theme" : "default" }}%%
graph TB
    User["user.ts<br/>Task interface"]:::major-edit
    TimeUtils["timeUtils.ts<br/>computeNextDueDate()"]:::major-edit
    AddTask["AddTask.tsx<br/>Create recurring tasks"]:::major-edit
    EditTask["EditTask.tsx<br/>Edit recurrence"]:::major-edit
    TaskMenu["TaskMenu.tsx<br/>Single completion"]:::major-edit
    TasksList["TasksList.tsx<br/>Bulk completion"]:::major-edit
    
    User --> AddTask
    User --> EditTask
    TimeUtils --> TaskMenu
    TimeUtils --> TasksList
    TaskMenu --> User
    TasksList --> User
    
    AddTask -.->|"creates tasks with<br/>recurrence field"| User
    EditTask -.->|"updates recurrence<br/>on existing tasks"| User
    TaskMenu -.->|"spawns next occurrence<br/>on single completion"| TimeUtils
    TasksList -.->|"spawns next occurrence<br/>on bulk completion"| TimeUtils

    subgraph Legend
        L1["Major Edit"]:::major-edit
        L2["Minor Edit"]:::minor-edit  
        L3["Context/No Edit"]:::context
    end

    classDef major-edit fill:#90EE90
    classDef minor-edit fill:#87CEEB
    classDef context fill:#FFFFFF
```

### Notes

This feature keeps the implementation simple as requested - no complex scheduling or notifications, just immediate next occurrence creation on completion. The monthly date clamping ensures no invalid dates (e.g., Jan 31 → Feb 28/29).

**Link to Devin run:** https://app.devin.ai/sessions/dd68d7be308943e792bc7470af7e6a05  
**Requested by:** Ethan Zerad (@ethanzrd)

### CODE REVIEW

File: src/components/tasks/TaskMenu.tsx [hunk 1]
```diff
@@ -28,6 +28,7 @@ import { UserContext } from "../../contexts/UserContext";
 import { useResponsiveDisplay } from "../../hooks/useResponsiveDisplay";
 import { Task } from "../../types/user";
 import { calculateDateDifference, generateUUID, showToast } from "../../utils";
+import { computeNextDueDate } from "../../utils/timeUtils";
```
Comment (ethanzrd) [inline src/components/tasks/TaskMenu.tsx:31]: what is this?
